js/ui/html_editor/modules/tableContextMenu.js

```diff
@@ -0,0 +1,118 @@
+import Quill from 'devextreme-quill';
+import $ from '../../../core/renderer';
+import BaseModule from './base';
+import eventsEngine from '../../../events/core/events_engine';
+import { addNamespace } from '../../../events/utils/index';
+import ContextMenu from '../../context_menu';
+import { getTableOperationHandler } from './tableOperations';
+
+
+const MODULE_NAMESPACE = 'dxHtmlTableContextMenu';
+
+const CONTEXT_MENU_EVENT = addNamespace('dxcontextmenu', MODULE_NAMESPACE);
+
+let TableContextMenuModule = BaseModule;
+
+if(Quill) {
+    TableContextMenuModule = class TableContextMenuModule extends BaseModule {
+        constructor(quill, options) {
+            super(quill, options);
+            this.enabled = !!options.enabled;
+            this._quillContainer = this.editorInstance._getQuillContainer();
+
+            if(this.enabled) {
+                this._contextMenu = this._createContextMenu();
+                this._attachEvents();
+            }
+        }
+
+        _attachEvents() {
+            eventsEngine.on(this.editorInstance._getContent(), CONTEXT_MENU_EVENT, this._openTableContextMenu.bind(this));
+        }
+
+        _detachEvents() {
+            eventsEngine.off(this.editorInstance._getContent(), MODULE_NAMESPACE);
+        }
+
+        _createContextMenu() {
+            const $container = $('<div>').appendTo(this.editorInstance.$element());
+            const menuConfig = this._getMenuConfig();
+
+            return this.editorInstance._createComponent($container, ContextMenu, menuConfig);
+        }
+
+        _getMenuConfig(options) {
+            return {
+                target: this._quillContainer,
+                showEvent: '',
+                dataSource: [
+                    { text: 'Insert', items: [
+                        { text: 'Insert Row Above', onClick: getTableOperationHandler(this.quill, 'insertRowAbove') },
```

Comment:
```suggestion
                        { text: 'Insert Header Row', onClick: getTableOperationHandler(this.quill, 'insertHeaderRow') },
                        { text: 'Insert Row Above', onClick: getTableOperationHandler(this.quill, 'insertRowAbove') },
```
Note: add\use localization constants for text